test/e2e/service.go

```diff
@@ -455,3 +447,10 @@ func addEndpointPodOrFail(c *client.Client, ns, name string, labels map[string]s
 	_, err := c.Pods(ns).Create(pod)
 	Expect(err).NotTo(HaveOccurred())
 }
+
+// dateStamp returns the current time as a string "YYYY-MM-DDTHHMMSS"
+// Handy for unique names across test runs
+func dateStamp() string {
```

Comment:
Agreed.  I decided to keep everything inside this test for now, just to verify that it stabilizes Jenkins e2e over the weekend.  I plan to clean all of this up nicely across all e2e tests soon, but that is a bigger and lower priority item.